### PR TITLE
Allow 'dw' and 'dl' to apply to characters of strings

### LINKED ISSUE #568: Allow 'dw' to apply to characters of strings
This is orthogonal to widechar support (#97); it's potentially useful even if charmaps remain 8-bit.

`db "ABC", -1, "{dee}EF"` with `charmap "{dee}", $42` outputs 7 bytes, since each character post-charmap lookup gets handled by `db`.

`dw "ABC", -1, "{dee}EF"` could output 14 bytes, just as if it had been written `dw "A", "B", "C", -1, "{dee}", "E", "F"`.

This can't be done with a macro, because even if the new lexer resolves the current issues around quotes and commas (consider `mydwstring "AB, C{dee}", "EF, GH"`), macro arguments would still see the individual characters `{` `d` `e` `e` `}`.

Currently:
- `dw "...PQ"` acts like `db "Q", "P"`
- `dw "X"` works as expected
- `dw ""` acts like `dw 0`, although `db ""` just outputs no bytes

### PR DESCRIPTION
**Pull request by Rangi42**
Fixes #568

The old behavior of `dw "string"` can be replicated with `dw ("string")`; likewise for dl

### CODE REVIEW

**Comment by Rangi42** [comment]
This would naturally extend to multi-byte charmaps if those ever become a thing; so if `"Œ"` were `$152` and `"℈"` were `$2108`, then `dw "Œ℈"` would output `52 01 08 21`.

**Comment by Rangi42** [comment]
(This feature was motivated by https://github.com/mid-kid/pokepicross/, which currently needs [this workaround macro](https://github.com/mid-kid/pokepicross/blob/master/include/text.inc).)

**Comment by ISSOtm** [comment]
I added documentation of this change, for review

File: src/asm/rgbasm.5 [hunk 1]
```diff
@@ -1075,28 +1075,34 @@ command-line option, except when using overlays with
...
+Strings are handled a little specially: they first undergo charmap conversion (see
+.Sx Character maps ) ,
+then each resulting character is output individually.
+For example, under the default charmap, the following two lines are identical:
+.Bd -literal -offset indent
+DW "Hello!"
+DW "H", "e", "l", "l", "o", "!"
+.Ed
+.Pp
+If you do not want this, enclose the string in parentheses.
```
Comment (Rangi42) [inline src/asm/rgbasm.5:1097]: "If you do not want this special handling", just to be clear about what the parentheses disable.

**Pull request merged**